lib/output/writer/elasticsearch.go

```diff
@@ -307,36 +309,49 @@ func (e *Elasticsearch) Write(msg types.Message) error {
 			return err
 		}
 
-		failed := result.Failed()
-		if len(failed) == 0 {
+		if !result.Errors {
+			requests = requests[:len(result.Items)]
 			continue
 		}
 
 		wait := boff.NextBackOff()
-		for i := 0; i < len(failed); i++ {
-			reason := "no reason given"
-			if fErr := failed[i].Error; fErr != nil {
-				reason = fErr.Reason
-			}
-			if !shouldRetry(failed[i].Status) {
-				e.log.Errorf("Elasticsearch message '%v' rejected with code [%v]: %v\n", failed[i].Id, failed[i].Status, reason)
-				return fmt.Errorf("failed to send %v parts from message: [%v]: %v", len(failed), failed[i].Status, reason)
-			}
-			e.log.Errorf("Elasticsearch message '%v' failed with code [%v]: %v\n", failed[i].Id, failed[i].Status, reason)
-			id := failed[i].Id
-			req := requests[id]
-			bulkReq, err := e.buildBulkableRequest(id, req)
-			if err != nil {
-				return err
+		errorCount := 0
+		var returnedErr *elastic.ErrorDetails
+		for i, resp := range result.Items {
+			for _, item := range resp {
+				if item.Status < 200 || item.Status > 299 {
+					if returnedErr == nil {
+						reason := "no reason given"
+						if item.Error != nil {
+							if i == 0 {
+								returnedErr = item.Error
+							}
+							reason = item.Error.Reason
+						}
+						if !shouldRetry(item.Status) {
+							e.log.Errorf("Elasticsearch message '%v' rejected with code [%v]: %v\n", item.Id, item.Status, reason)
+							returnedErr = item.Error
```

Comment:
previously this was returning. but since we need the error count (since that was used previously) I just use `returnedErr` to effectively do the same thing after we have counted all the errors